pkgs/build-support/singularity-tools/default.nix

```diff
@@ -85,8 +92,8 @@ lib.makeExtensible (final: {
             layerClosure = writeClosure (
               [
                 bashInteractive
-                runScriptFile
               ]
+              ++ runScriptReferences
```

Comment:
I think I finally get it: your aim is to get `runScriptFile`'s dependencies into the image, without holding two copies of the runscript (one in `.singularity.d` and one in `/nix/store`)?